frontend/simple/js/backend/hapi.js

```diff
@@ -76,7 +76,11 @@ export class HapiBackend extends Backend {
   async subscribe (contractId: string) {
     console.log('subscribing to:', contractId)
     // this mutation makes sure not to add duplicates
-    store.commit('pending', contractId)
+    // don't add this to pending if its already confirmed
+    // this can happen when loading and resubscribing
+    if (!store.state.contracts[contractId]) {
+      store.commit('pending', contractId)
+    }
```

Comment:
This should be done not here, but in the mutation itself, similarly to how the mutation checks if it's already pending.

(Try to use the same 1-liner style that's there to maintain consistency and Save The Lions `index === -1 && state.pending.push(contractId)`.)